packages/babel-traverse/src/path/family.js

```diff
@@ -81,14 +81,21 @@ export function getCompletionRecords(): NodePath[] {
     paths = addCompletionRecords(this.get("alternate"), paths);
   } else if (this.isDoExpression() || this.isFor() || this.isWhile()) {
     paths = addCompletionRecords(this.get("body"), paths);
-  } else if (this.isProgram() || this.isBlockStatement()) {
+  } else if (this.isProgram()) {
+    paths = addCompletionRecords(this.get("body").pop(), paths);
+  } else if (this.isBlockStatement()) {
```

Comment:
We can not remove `BreakStatement` because it will lose the completion record when the `prevSibling` is neither an ExpressionStatement nor a BlockStatement. i.e.
```js
do { switch(0) { case 0: {
  if (true) { break; } "unreachable"; 
  } }
}
```
If `break` is removed, `"unreachable"` will be incorrectly included in the completion record.

The completion record tracking of SwitchStatement was introduced at #10070. The `BreakStatement` is handled in https://github.com/babel/babel/blob/e498bee10f0123bb208baa228ce6417542a2c3c4/packages/babel-traverse/src/path/family.js#L52-L60

The `BreakStatement` is searched in `consequent` and `consequent[type=BlockStatement] > body`

https://github.com/babel/babel/blob/e498bee10f0123bb208baa228ce6417542a2c3c4/packages/babel-traverse/src/path/family.js#L29-L41

Apparently this routine can not find `BreakStatement` nested in `IfStatement` of the `SwitchCase`. We can do a tree search for `BreakStatement`, which resembles `getCompletionRecords`.

```js
function findBreak(statements) {
  let breakStatement;
  if (!Array.isArray(statements)) {
    statements = [statements];
  }

  for (const statement of statements) {
    if (statement.isBlockStatement()) {
      breakStatement = findBreak(statement.get("body"));
    } else if (statement.isIfStatement()) {
      breakStatement =
        findBreak(statement.get("consequent")) ??
        findBreak(statement.get("alternate"));
    } else if (statement.isCatchClause() || statement.isLabeledStatement()) {
      breakStatement = findBreak(statement.get("body"));
    // todo: add other statements except Iteration and Switch, which is breakable statement.
    } else if (statement.isBreakStatement()) {
      breakStatement = statement;
    }
    if (breakStatement) {
      return breakStatement;
    }
  }
  return null;
}
```
By doing so we can reuse current `completionRecordForSwitch` routine which already handled `BreakStatement`.

Reply:
What is the reasoning behind returning `getCompletionRecords` as opposed to appending to `paths` in the `isFunction` case?

Reply:
This is a good question! Each function owns a new execution context. So when a `Function` is met, we should skip finding `break` inside the function body because the statements inside can not interfere with completion records (control flows) outside. i.e.

```js
{
  a: 42;
  () => { break a }
}
```
is invalid because the label `a` is _not_ defined in the execution context own by the arrow function.

Reply:
That completely makes sense. I have made the necessary changes. Please let me know if there needs to be any others!